File: qzdb/java/src/main/java/com/qqzeng/qzdb/RowIds.java
package com.qqzeng.qzdb;

/**
 * 对应 IP 行中引用的 ID 结构体 (用于底层 lookupIds)
 *
 * @param geoId   地理位置 ID
 * @param asnId   ASN ID
 * @param usageId 应用场景 ID
 */
public record RowIds(int geoId, int asnId, int usageId) {
}

File: ip-qzdb-sdk/netcore/RowIds.cs
namespace QQZeng.Qzdb;

/// <summary>Reverse-lookup row identifiers for a resolved IP entry. Legacy compatibility struct; the canonical API returns a named tuple from <see cref="QzdbReader.LookupIds(uint)"/>.</summary>
/// <param name="GeoId">Index into the geo string pool for this entry.</param>
/// <param name="AsnId">Index into the ASN string pool (0 when the group has no ASN dimension).</param>
/// <param name="UsageId">Index into the usage-type string pool (0 when the group has no usage dimension).</param>
public readonly record struct RowIds(int GeoId, int AsnId, int UsageId);
